panic: assignment to entry in nil map

## Bug Report

### Description

```
./talosctl apply-config --insecure --nodes 192.168.1.180 --interactive
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/talos-systems/talos/pkg/machinery/client/config.(*Config).Merge(0xc000afa400, 0xc00012e060, 0x2, 0x2, 0x3)
	/src/pkg/machinery/client/config/config.go:148 +0x31d
github.com/talos-systems/talos/internal/pkg/tui/installer.(*Installer).writeTalosconfig(0xc0004a8ea0, 0xc000530180, 0xc00012e060, 0x0, 0x0)
	/src/internal/pkg/tui/installer/installer.go:421 +0x866
github.com/talos-systems/talos/internal/pkg/tui/installer.(*Installer).apply(0xc0004a8ea0, 0xc0001105a0, 0x0, 0x0)
	/src/internal/pkg/tui/installer/installer.go:380 +0x53f
github.com/talos-systems/talos/internal/pkg/tui/installer.(*Installer).Run(0xc0004a8ea0, 0xc0001105a0, 0x0, 0x0)
	/src/internal/pkg/tui/installer/installer.go:95 +0x345
github.com/talos-systems/talos/cmd/talosctl/cmd/talos.glob..func1.2(0x2c77820, 0xc000050088, 0xc0000cc240, 0x2, 0x2)
	/src/cmd/talosctl/cmd/talos/apply-config.go:133 +0x110
github.com/talos-systems/talos/cmd/talosctl/cmd/talos.glob..func1.1(0xc00029b440, 0x0, 0x0)
	/src/cmd/talosctl/cmd/talos/apply-config.go:94 +0x3f2
github.com/talos-systems/talos/cmd/talosctl/cmd/talos.glob..func1(0x38f11c0, 0xc000710740, 0x0, 0x4, 0x0, 0x0)
	/src/cmd/talosctl/cmd/talos/apply-config.go:100 +0xc2
github.com/spf13/cobra.(*Command).execute(0x38f11c0, 0xc000710680, 0x4, 0x4, 0x38f11c0, 0xc000710680)
	/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:850 +0x47c
github.com/spf13/cobra.(*Command).ExecuteC(0x38ee7c0, 0x390d6d0, 0x29ef79c, 0x9)
	/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:958 +0x375
github.com/talos-systems/talos/cmd/talosctl/cmd.Execute(0x0, 0x0)
	/src/cmd/talosctl/cmd/root.go:42 +0x235
main.main()
	/src/cmd/talosctl/main.go:18 +0x33
```

Built from git tag `v0.9.0-alpha.0`.

That happens when talosconfig file is present, but empty or contains only `---`.

--- a/pkg/machinery/client/config/config.go
+++ b/pkg/machinery/client/config/config.go
@@ -123,6 +123,10 @@ func (r *Rename) String() string {
 //
 // Current context is overridden from passed in config.
 func (c *Config) Merge(cfg *Config) []Rename {
+	if c.Contexts == nil {
+		c.Contexts = map[string]*Context{}
+	}
+
 	mappedContexts := map[string]string{}
 	renames := []Rename{}
 

--- a/pkg/machinery/client/config/config_test.go
+++ b/pkg/machinery/client/config/config_test.go
@@ -25,10 +25,8 @@ func TestConfigMerge(t *testing.T) {
 		expectedContexts map[string]*config.Context
 	}{
 		{
-			name: "IntoEmpty",
-			config: &config.Config{
-				Contexts: map[string]*config.Context{},
-			},
+			name:   "IntoEmpty",
+			config: &config.Config{},
 			configToMerge: &config.Config{
 				Context: "foo",
 				Contexts: map[string]*config.Context{

